### PR TITLE
[DDW-1093] Update cardano-js to v0.2.2 and address validator improvement

### PR DESCRIPTION
**Pull request by tomothespian**
This PR introduces cardano-js package updated to version 0.2.2 and address validation improvement

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1574355883105700)
- [ ] Go to wallet send screen and enter receiver address to check if address validation works properly.

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/stores/WalletsStore.js [hunk 1]
```diff
@@ -639,19 +639,11 @@ export default class WalletsStore extends Store {
   };
 
   isValidAddress = (address: string) => {
-    const { app, networkStatus } = this.stores;
-    const { environment } = app;
-    const { network } = environment;
-    const { nodeImplementation } = networkStatus;
+    const { app } = this.stores;
+    const { isMainnet } = app.environment;
+    const chainSettings = isMainnet ? MAINNET : TESTNET;
```
[the remark this answers is not in the document]
Comment (rhyslbw) [reply source/renderer/app/stores/WalletsStore.js]: These are discriminants or _network magic_,  built into the address format:
See:
https://github.com/input-output-hk/cardano-wallet/wiki/About-Address-Format---Byron
and:
https://github.com/input-output-hk/implementation-decisions/blob/master/text/0001-address.md

**Comment by tomothespian** [comment]
@nikolaglumac I don't see any flow issue :(
Screen Shot 2019-11-22 at 10 02 44

File: source/renderer/app/stores/WalletsStore.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhyslbw) [reply source/renderer/app/stores/WalletsStore.js]: These are less about specific networks, more to indicate the type of network, so in the case of Jörmungandr, eveything so far should be considered `testnet`

**Comment by tomothespian** [comment]
@rhyslbw @nikolaglumac This one is updated! Can you please take a look again? 🙏

**Pull request merged**